GoogleUtilities/Environment/GULHeartbeatDateStorage.m

```diff
@@ -20,10 +20,14 @@
 @interface GULHeartbeatDateStorage ()
 /** The storage to store the date of the last sent heartbeat. */
 @property(nonatomic, readonly) NSFileCoordinator *fileCoordinator;
+/** The name of the file that stores heartbeat information. */
+@property(nonatomic, readonly) NSString *fileName;
 @end
 
 @implementation GULHeartbeatDateStorage
 
+@synthesize fileURL = _fileURL;
```

Comment:
nit: Is it required? We usually use implicit `synthesize` unless it's not possible or lead to errors.

Reply:
Yes, since the property is `readonly` and we implement it's only required accessor (the getter), it isn't auto synthesized so we need to add an implicit `synthesize`.

Reply:
Is it related to overriding the getter of a readonly property? Interesting, I didn't know it's the case.

Reply:
Yup, I found this in apple's objc docs about [properties](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/EncapsulatingData/EncapsulatingData.html):

>Note: The compiler will automatically synthesize an instance variable in all situations where it’s also synthesizing at least one accessor method. If you implement both a getter and a setter for a readwrite property, or a getter for a readonly property, the compiler will assume that you are taking control over the property implementation and won’t synthesize an instance variable automatically.

I never knew that was the reason but it makes sense I guess!